app/src/main/java/com/example/gradeassist/service/IScoreService.kt

```diff
@@ -1,8 +1,36 @@
 package com.example.gradeassist.service
 
+import androidx.lifecycle.MutableLiveData
+import com.example.gradeassist.RetrofitClientInstance
+import com.example.gradeassist.dao.IScoreDAO
 import com.example.gradeassist.dto.Score
+import retrofit2.Call
+import retrofit2.Callback
+import retrofit2.Response
 
 public interface IScoreService {
 
-    fun fetchById(scoreId: Int) : Score
+    fun fetchScoreById(scoreId: Int) : MutableLiveData<Score>{
+        var _score = MutableLiveData<Score> ()
+        val service = RetrofitClientInstance.retrofitInstance?.create(IScoreDAO::class.java)
+        service?.getScoreById(scoreId)
+        val call = service?.getScoreById(scoreId)
+        call?.enqueue(object: Callback<Score> {
+            override fun onResponse(
+                call: Call<Score>,
+                response: Response<Score>
+            ) {
+                _score.value = response.body()
+            }
+
+            override fun onFailure(call: Call<Score>, t: Throwable) {
+               // TODO finish onFailure function
+            }
+
+        })
+
+        return _score
+
+
+    }
 }
```

Comment:
Added a new interface for service (IScoreService) with a function that fetches scores by id with retrofit to parse JSON

Reply:
MutableLiveData is a concept that belongs in a ViewModel, not necessarily a Service class.  The ViewModel can call the Service class and wrap the results into MutableLiveData, but I'd lean toward having the Service class return a plain old object or collection.

Reply:
Thank you for the clarifications, this is all new to me.